app/models/movie_search.rb

```diff
@@ -13,30 +14,27 @@ def initialize(title, in_db, tmdb_id, release_date, vote_average, overview, back
     @lists = lists
   end
 
-  attr_accessor :title, :in_db, :tmdb_id, :release_date, :vote_average, :overview, :backdrop_path, :poster_path, :tags, :lists
-
-  def self.parse_results(json)
-    @movies = []
-    json.each do |result|
-      @tmdb_id = result[:id]
-      @in_db = Movie.exists?(tmdb_id: @tmdb_id)
-        if @in_db
-          @movies << Movie.find_by(tmdb_id: @tmdb_id)
-        else
-          @title = result[:title]
-          @release_date = result[:release_date]
-          @vote_average = result[:vote_average]&.round(1)
-          @overview = result[:overview]
-          @backdrop_path = result[:backdrop_path]
-          @poster_path = result[:poster_path]
-          @tags = nil
-          @lists = nil
+  def self.parse_results(results)
+    results.map do |result|
+      tmdb_id = result[:id]
+      existing_movie = Movie.find_by(tmdb_id: @tmdb_id)
 
-          @movie = MovieSearch.new(@title, @in_db, @tmdb_id, @release_date, @vote_average, @overview, @backdrop_path, @poster_path, @tags, @lists)
-          @movies << @movie
-        end
-    end #loop
-    @movies
+      if existing_movie.present?
+        existing_movie
+      else
+        new(
```

Comment:
It seems like there is probably a more elegant way to do this, but my brain got tired and this is how I left it. Any suggestions?

Reply:
I mean maybe `Movie.find_by(tmdb_id: @tmdb_id) || new(` but I think the way you currently have it is more readable?

Reply:
I'm beginning to wonder why we're making a different object here and not just instantiating a `Movie` object. This `Movie` would probably not ever be saved. I know that later downstream in the `app/views/movies/_movie_partial_loop.html.erb` we ask the object it's class for the sake of routing:
```
<% if movie.is_a?(Movie) %>
  <%= link_to image_for(movie), movie_path(movie) %>
<% else %>
  <%= link_to image_for(movie), movie_more_path(tmdb_id: movie.tmdb_id) %>
<% end %>
```
so I'm thought-swirling a bit on how to do this whole concept in a better way. Not for this PR -- that would be a different refactor.

Reply:
Yeah, sounds good for a different PR 👍